APCAppCore/APCAppCore/Library/DataArchiverAndUploader/APCDataArchive.m

```diff
@@ -0,0 +1,267 @@
+    }
+}
+
+//A sandbox in the temporary directory for this archive to be cleaned up on completion.
+- (NSString *)workingDirectoryPath
+{
+    
+    NSString *workingDirectoryPath = [NSTemporaryDirectory() stringByAppendingPathComponent:self.reference];
+    
+    if (![[NSFileManager defaultManager] fileExistsAtPath:workingDirectoryPath]) {
+        NSError * fileError;
+        BOOL created = [[NSFileManager defaultManager] createDirectoryAtPath:workingDirectoryPath withIntermediateDirectories:YES attributes:@{ NSFileProtectionKey : NSFileProtectionComplete } error:&fileError];
+        if (!created) {
+            APCLogError2 (fileError);
+        }
+    }
+    
+    return workingDirectoryPath;
+}
+
+- (void)insertDataAtURLIntoArchive: (NSURL*) url fileName: (NSString *) filename extension:(NSString *)extension
+{
+    NSString *fullFilename = [filename stringByAppendingPathExtension:extension];
+
+    [self.zipEntries addObject: [ZZArchiveEntry archiveEntryWithFileName: fullFilename
+                                                                compress:YES
+                                                               dataBlock:^(NSError** error)
+                                 {
+                                     APCLogError2(*error);
+                                     return [NSData dataWithContentsOfURL:url];
+                                 }]];
+    
+    //add the fileInfoEntry
+    NSDictionary *fileInfoEntry = @{ kFileInfoNameKey: fullFilename,
+                                     kFileInfoTimeStampKey: [NSDate date].toStringInISO8601Format,
+                                     kFileInfoContentTypeKey: [self contentTypeForFileExtension:extension] };
+    
+    [self.filesList addObject:fileInfoEntry];
+}
+
+- (void)insertJSONDataIntoArchive:(NSData *)jsonData filename:(NSString *)filename
+{
+    
+    if (jsonData !=nil) {
+        NSString *extension = @"json";
+        NSString * fullFileName = [filename stringByAppendingPathExtension:extension];
+        
+        APCLogFilenameBeingArchived (fullFileName);
+        
+        id objectToInsert = [ZZArchiveEntry archiveEntryWithFileName: fullFileName
+                                                            compress:YES
+                                                           dataBlock:^(NSError** __unused error){ return jsonData;}];
+        
+        if (objectToInsert) {
+            [self.zipEntries addObject: objectToInsert];
+            
+            NSDictionary *fileInfoEntry = @{ kFileInfoNameKey: fullFileName,
+                                             kFileInfoTimeStampKey: [NSDate date].toStringInISO8601Format,
+                                             kFileInfoContentTypeKey:[self contentTypeForFileExtension:extension] };
+            
+            [self.filesList addObject:fileInfoEntry];
+            
+        }else {
+            APCLogDebug(@"failed to insert %@ into archive", filename);
+        }
+    }
+}
```

Comment:
Consolidate `insertDataAtURLIntoArchive:` and `insertJSONDataIntoArchive:`, I see many common code.